src/message.spec.ts

```diff
@@ -127,3 +127,125 @@ test('self()', t => {
+
+  const ROOM_ID = '@@9cdc696e490bd76c57e7dd54792dc1408e27d65e312178b1943e88579b7939f4'
+
+  // Mock
+  const mockContactGetter = function (id) {
+    return new Promise((resolve, reject) => {
+      if (id !== ROOM_ID && !(id in CONTACT_LIST)) return resolve({})
+      if (id === ROOM_ID) {
+        resolve(RAW_OBJ)
+      }
+      if (id in CONTACT_LIST) {
+        setTimeout(() => {
+          return resolve(CONTACT_LIST[id])
+        }, 10)
+      }
+    })
+  }
+
+  let puppet1
+  try {
+    puppet1 = Config.puppetInstance()
+    puppet1.getContact = mockContactGetter
+  } catch (err) {
+    puppet1 = { getContact: mockContactGetter }
+    Config.puppetInstance(puppet1)
+  }
+  const msg11 = new Message(rawObj11)
+  const room11 = msg11.room()
+  if (room11) {
+    await room11.ready()
+    setTimeout(function () {
```

Comment:
You'd better find out why you need `setTimeout` at here, or you will be more likely to be bitten again and again.